Shared/Samples/Add WFS layer/AddWFSLayerView.swift

```diff
@@ -0,0 +1,125 @@
+// Copyright 2025 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+@MainActor
+struct AddWFSLayerView: View {
+    /// A map with a topographic basemap centered on downtown Seattle.
+    @State private var map: Map = {
+        let map = Map(basemapStyle: .arcGISTopographic)
+        map.initialViewpoint = Viewpoint(
+            boundingGeometry: Envelope(
+                min: Point(x: -122.341581, y: 47.617207, spatialReference: .wgs84),
+                max: Point(x: -122.332662, y: 47.613758, spatialReference: .wgs84)
+            )
+        )
+        return map
+    }()
+    
+    /// The visible area on the map.
+    @State private var visibleArea: ArcGIS.Polygon?
+    
+    /// A feature table of building footprints for downtown Seattle.
+    private let featureTable: WFSFeatureTable = {
+        let featureTable = WFSFeatureTable(url: .downtownSeattle, tableName: "Seattle_Downtown_Features:Buildings")
+        // Sets the feature request mode to manual. In this mode, the table must be populated
+        // manually. Panning and zooming won't request features automatically.
+        featureTable.featureRequestMode = .manualCache
+        // Sets the axis order.
+        featureTable.axisOrder = .noSwap
+        return featureTable
+    }()
+    
+    /// A Boolean value indicating whether the feature table is being populated.
+    @State private var isPopulating = true
+    
+    /// The error shown in the error alert.
+    @State private var error: Error?
+    
+    var body: some View {
+        MapView(map: map)
+            .onVisibleAreaChanged { visibleArea = $0 }
+            .onNavigatingChanged { isNavigating in
+                guard !isNavigating else { return }
+                Task {
```

Comment:
this unstructured task will not get cancelled. It would be better to have a state variable for `isNavigating` and then use a .task modifier to run when isNavigating changes. That makes it a task that SwiftUI manages the lifetime of and will cancel previous tasks when it runs a new one.